functions/Get-DbaFirewallRule.ps1

```diff
@@ -0,0 +1,235 @@
+            }
+
+            # If command was not successful, just output messages and continue with next SqlInstance
+            if (-not $commandResult.Successful) {
+                [PSCustomObject]@{
+                    ComputerName = $instance.ComputerName
+                    Warning      = $commandResult.Warning
+                    Error        = $commandResult.Error
+                    Exception    = $commandResult.Exception
+                    Details      = $commandResult
+                } | Select-DefaultView -Property ComputerName, Warning, Error, Exception
+                continue
+            }
+
+            # Add more information to the rules
+            $rules = foreach ($rule in $commandResult.Rules) {
+                if ($rule.Name -eq 'SQL Server Browser') {
+                    $typeName = 'Browser'
+                    $instanceName = $null
+                    $sqlInstanceName = $null
+                } elseif ($rule.Name -eq 'SQL Server default instance') {
+                    $typeName = 'Engine'
+                    $instanceName = 'MSSQLSERVER'
+                    $sqlInstanceName = $instance.ComputerName
+                } else {
+                    $typeName = 'Engine'
+                    $instanceName = $rule.Name -replace '^SQL Server instance (.+)$', '$1'
+                    $sqlInstanceName = $instance.ComputerName + '\' + $instanceName
+                }
+                [PSCustomObject]@{
+                    ComputerName = $instance.ComputerName
+                    InstanceName = $instanceName
+                    SqlInstance  = $sqlInstanceName
+                    DisplayName  = $rule.DisplayName
+                    Name         = $rule.Name
+                    Type         = $typeName
+                    Protocol     = $rule.Protocol
+                    LocalPort    = $rule.LocalPort
+                    Program      = $rule.Program
+                    Rule         = $rule
+                    Credential   = $Credential
+                }
+            }
+
+            # What rules should we output?
+            $outputRules = @( )
+            if ('AllOnComputer' -in $Type) {
+                Write-Message -Level Verbose -Message 'Returning all rules for target computer'
+                $outputRules += $rules
+            } elseif ($null -eq $Type) {
+                Write-Message -Level Verbose -Message 'Returning rule for instance and maybe for Browser'
+                # Get the rule for the instance
+                $outputRules += $rules | Where-Object { $_.Type -eq 'Engine' -and $_.InstanceName -eq $instance.InstanceName }
+                if ($outputRules.Count -eq 0) {
+                    Write-Message -Level Verbose -Message 'No rule found for instance'
+                } elseif ($outputRules.LocalPort -eq '1433') {
```

Comment:
I didn't see any of this in my output, and I'm left wondering -- did we do it by TCP port or Program? Can the output be more useful?

Reply:
It depends on the network configuration of the instance: Static port uses LocalPort, dynamic port uses Program.
What do you mean by "more useful"?

Reply:
I'd like to see what it did. If it picked a static port, I'd like to see that port 1433 was secured. If it uses Program, I'd like to see the path.

Reply:
Oh, now I understand. The Get- only works by Name of the rule. So we don't need to get the configuration of the instance. Only New- needs to get the configuration to build the rule. The LocalPort and Program is then part of the output.

Reply:
Oh oops, yes it was New. Can we get the details from Get?

Reply:
What is missing?
![image](https://user-images.githubusercontent.com/66946165/126311510-b121a69a-4733-4ee3-b58a-09ea78da8df3.png)

Reply:
Oh weird. Nothing on yours. I may have had an old commit, my apologies. That is beautiful, can we get something as detailed for all DbaFirewallRule commands?